### DISCUSSION TITLE
Is there support for SCRIPT_NAME?

### DISCUSSION THREAD

**Discussion by Midnighter** [fastapi/fastapi#8030] [Questions] [answered]
**Description**

Is the use of `SCRIPT_NAME` is fully supported in FastAPI? Or is there a different strategy for mounting the main application on a URL path?

**Additional context**

I have created a [demo project](https://github.com/Midnighter/fastapi-mount) with an extensive explanation of all the steps I tried.

I'm deploying my applications with gunicorn and uvicorn. So it is possibly also a uvicorn issue. Whereas [daphne has added support](https://github.com/django/daphne/issues/24) for `SCRIPT_NAME`, I can find no mention of it in the uvicorn docs or issues.

**Comment by Midnighter** [comment]
I found a solution but would still request a change to FastAPI. I'll edit this comment in a bit. I need to deploy the solution before explaining it here :wink:

**Comment by Midnighter** [comment]
So I found out that I need to configure the `root_path` rather than the `SCRIPT_NAME`. You can read up on [the full solution strategy on the root-path branch](https://github.com/Midnighter/fastapi-mount/tree/root-path#solution). Which basically answers this question.

Follow ups:
1. I learnt a thing or two about proxying with NGINX and traefik, so I could add that to the docs somewhere.
2. I think that instead of manually configuring the `openapi_prefix` for the main app, it should automatically pick up on a configured `root_path` and use that (in addition to what is manually set). If you agree, I can try to make a PR.

Thanks to @euri10 for the discussion on gitter.

**Comment by tiangolo** [comment]
@Midnighter I'm glad you solved your issue!

You might want to check the docs for using apps behind a proxy: https://fastapi.tiangolo.com/advanced/sub-applications-proxy/

**Comment by Midnighter** [comment]
I did look at that section a fair bit.

What do you think about my suggestion above?

> I think that instead of manually configuring the `openapi_prefix` for the main app, it should automatically pick up on a configured `root_path` and use that (in addition to what is manually set). If you agree, I can try to make a PR.

**Comment by Midnighter** [comment]
Bump.

To me it makes sense to automatically include the root path as an openapi prefix and I'm happy to make the required changes.

**Comment by tiangolo** [comment, marked as answer]
Hey @Midnighter ! There's now support for `root_path` in FastAPI :tada:  https://fastapi.tiangolo.com/advanced/sub-applications/

**Comment by Midnighter** [comment]
I was happy to see it in the latest release notes! Very cool :slightly_smiling_face:
